lms/djangoapps/instructor/views/api.py

```diff
@@ -3014,3 +3041,141 @@ def build_row_errors(key, _user, row_count):
     }
 
     return JsonResponse(results)
+
+
+@transaction.non_atomic_requests
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_global_staff
+@require_http_methods(['POST', 'DELETE'])
+def certificate_invalidation_view(request, course_id):
+    """
+    Invalidate/Re-Validate students to/from certificate.
+
+    :param request: HttpRequest object
+    :param course_id: course identifier of the course for whom to add/remove certificates exception.
+    :return: JsonResponse object with success/error message or certificate invalidation data.
+    """
+    course_key = CourseKey.from_string(course_id)
+    # Validate request data and return error response in case of invalid data
+    try:
+        certificate_invalidation_data, certificate = validate_request_data_and_get_certificate(request, course_key)
```

Comment:
instead of passing the `request` object in function, can we use the `parse_request_data(request)` here and pass it to this and `validate_request_data_and_get_certificate` just return the one item `certificate` object ?